File: app/src/main/java/net/opendasharchive/openarchive/db/ProjectAdapter.java
package net.opendasharchive.openarchive.db;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.text.SpannableString;
import android.text.style.ImageSpan;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import net.opendasharchive.openarchive.R;
import net.opendasharchive.openarchive.fragments.MediaGridFragment;
import net.opendasharchive.openarchive.fragments.MediaListFragment;
import net.opendasharchive.openarchive.fragments.NewProjectFragment;
import net.opendasharchive.openarchive.projects.AddProjectActivity;
import net.opendasharchive.openarchive.util.SmartFragmentStatePagerAdapter;

import java.util.List;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

import static net.opendasharchive.openarchive.MainActivity.REQUEST_NEW_PROJECT_NAME;

/**
 * Created by micahjlucas on 1/20/15.
 */
public class ProjectAdapter extends SmartFragmentStatePagerAdapter {

    private List<Project> data;
    private Activity context;

    public ProjectAdapter(Activity context, FragmentManager fm) {
        super(fm);
        this.context = context;

    }

    @Override
    public Fragment getItem(int i) {

        if (i == 0)
        {
            NewProjectFragment frag = new NewProjectFragment();
            return frag;
        }
        else {
            MediaListFragment fragment = new MediaGridFragment();
            fragment.setProjectId(getProject(i).getId());
            Bundle args = new Bundle();
            fragment.setArguments(args);
            return fragment;
        }
    }

    public Project getProject (int i)
    {
        if (data != null && i > 0)
            return data.get(i-1);
        else
            return null;
    }

    public void updateData (List<Project> data)
    {
        this.data = data;
        notifyDataSetChanged();
    }

    @Override
    public int getCount() {
        if (data != null)
            return data.size()+1;
        else
            return 0;
    }

    @Override
    public CharSequence getPageTitle(int position) {

        if (position == 0)
        {
            ImageSpan imageSpan = new ImageSpan(context, R.drawable.ic_add_circle_outline_black_24dp);
            SpannableString spannableString = new SpannableString(" ");

            int start = 0;
            int end = 1;
            int flag = 0;
            spannableString.setSpan(imageSpan, start, end, flag);

            return spannableString;
        }
        else
            return data.get(position-1).description;
    }




}

File: app/src/main/java/net/opendasharchive/openarchive/db/ProjectAdapter.kt
package net.opendasharchive.openarchive.db

import android.content.Context
import android.os.Bundle
import android.text.SpannableString
import android.text.style.ImageSpan
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import net.opendasharchive.openarchive.R
import net.opendasharchive.openarchive.features.media.NewFolderFragment
import net.opendasharchive.openarchive.features.media.grid.MediaGridFragment
import net.opendasharchive.openarchive.features.media.list.MediaListFragment
import net.opendasharchive.openarchive.util.SmartFragmentStatePagerAdapter

class ProjectAdapter(private val context: Context, fragmentManager: FragmentManager) : SmartFragmentStatePagerAdapter(fragmentManager) {

    var projects = listOf<Project>()
        private set

    override fun getItem(position: Int): Fragment {
        return if (position == 0) {
            NewFolderFragment()
        }
        else {
            MediaGridFragment().also { fragment ->
                getProject(position)?.let { project ->
                    fragment.setProjectId(project.id)
                }

                fragment.arguments = Bundle()
            }
        }
    }

    override fun getCount(): Int {
        return projects.size + 1
    }

    fun getProject(i: Int): Project? {
        return if (i > 0 && i <= projects.size) projects[i - 1] else null
    }

    fun updateData(projects: List<Project>) {
        this.projects = projects

        notifyDataSetChanged()
    }

    fun getIndex(project: Project?): Int {
        val default = if (projects.isNotEmpty()) 1 else 0

        if (project == null) {
            return default
        }

        return (projects.indexOf(project) + 1).coerceAtLeast(default)
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return if (position == 0) {
            val imageSpan = ImageSpan(context, R.drawable.ic_add_circle_outline_black_24dp)

            val spannableString = SpannableString(" ")
            spannableString.setSpan(imageSpan, 0, 1, 0)

            spannableString
        }
        else {
            getProject(position)?.description
        }
    }

    fun getRegisteredMediaListFragment(position: Int): MediaListFragment? {
        return getRegisteredFragment(position) as? MediaListFragment
    }
}